Mathlib/Tactic/Recall.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2023 Mac Malone. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mac Malone, Kyle Miller
+-/
+import Lean.Elab.MutualDef
+import Std.Tactic.OpenPrivate
+
+/-!
+# `recall` command
+-/
+
+namespace Mathlib.Tactic.Recall
+
+/--
+The `recall` command redeclares a previous definition for illustrative purposes.
+This can be useful for files that give an expository account of some theory in Lean.
+
+The syntax of the command mirrors `def`, so all the usual bells and whistles work.
+```
+recall HasFDerivAtFilter (f : E → F) (f' : E →L[𝕜] F) (x : E) (L : Filter E) :=
+  (fun x' => f x' - f x - f' (x' - x)) =o[L] fun x' => x' - x
+```
+Also, one can leave out the body.
+```
+recall Nat.add_comm (n m : Nat) : n + m = m + n
+```
+
+The command verifies that the type and value provided are definitionally equal to the
+original declaration. However, this does not capture some details (like binders), so the
+following works without error.
+```
+recall Nat.add_comm {n m : Nat} : n + m = m + n
+```
+-/
+syntax (name := recall) "recall " ident ppIndent(optDeclSig) (declVal)? : command
+
+open Lean Meta Elab Command Term
+open private elabHeaders from Lean.Elab.MutualDef
+
+elab_rules : command
+  | `(recall $id $sig:optDeclSig $[$val?]?) => withoutModifyingEnv do
+    let some info := (← getEnv).find? id.getId
+      | throwError "unknown constant '{id}'"
+    let id' := mkIdentFrom id (← mkAuxName id.getId 1)
```

Comment:
If the lookup is successful, you should put TermInfo on the identifier so that go to definition works on it (and takes you to the original def, not the transitory one made here).

Reply:
Oh, I did not know you could do that. Will do!